### PR TITLE
Add automatic git tag and pypi deployment

### LINKED ISSUE #9: Add automatic pypi deployment and git tag
Add automatic deployment for `extension-template` repository. 
And add tag on git for deployed version.

See `country-template` [deployment script](https://github.com/openfisca/country-template/blob/e18b9d9cbaf02f4f4811e43c15e15e643b6bcb17/deploy-if-version-bump.sh).
 
> `extension-template` isn't on Cicle CI either.

### PR DESCRIPTION
**Pull request by sandcha**
Fixes #9 

Adds scripts to:
- add git tag on version bump
- auto-deploy on pypi on version bump

And move CircleCI config to v2
> Including setting new status checks on GitHub > Settings > Branches > master

Details : 
1. **GitHub repository events > CircleCI**  
On CircleCI `ADD PROJECTS`, followed extension-template repository. 
This adds an [openfisca/extension-template deploy key](https://circleci.com/gh/openfisca/extension-template/edit#checkout) on CicleCI and SSH [deploy key](https://github.com/openfisca/extension-template/settings/keys) on GitHub.

2. **CircleCI > Pypi**  
On CicleCI extension-template `Settings` > `Build Settings` > `Environment Variables`, imported `PYPI_PASSWORD` variable (CicleCI already knows its value from other repositories).

3. **CircleCI openfisca-bot > GitHub repository**  
To allow git tagging:  
As `openfisca-bot` logged on CicleCI and registered collaborator of `extension-template` on GitHub: added `openfisca-bot user key` on ["Checkout SSH keys" permissions page](https://circleci.com/gh/openfisca/extension-template/edit#checkout)  by clicking on : 
* Before 07.2020: `Authorize with GitHub` (Add user key) > `Create and add openfisca-bot user key`  
* Update on 07.2020: `Add user key` 
On openfisca-bot GitHub account, this adds a [CircleCI](https://github.com/settings/keys) key.
> If the repository isn't listed by CicleCI, it should be added on [this page](https://circleci.com/add-projects/gh/openfisca) (and followed).

4. **CircleCI openfisca-bot > Pypi**  
To allow pypi release:  
As [pypi](https://pypi.org/account/login/) user and owner of `extension-template`, added `openfisca-bot` as `Maintainer` in [extension-template collaborators](https://pypi.org/manage/project/openfisca-extension-template/collaboration/) 

> If pypi didn't know about the project, I would have added it a first time by: 
> 1. `python setup.py bdist_wheel`  # To create the package with a different revision (aka x.x.x.rc where rc stands for release candidate, ...) or by publishing it on pypi **test** url. 
⚠️ This should be done carefully as a published revision of a package on pypi can't be replaced/fixed later (see [pypi doc](https://pypi.org/help/#file-name-reuse) on file name reuse).
> 2. and `twine upload dist/* --username my_pypi_username`

### CODE REVIEW

**Comment by sandcha** [comment]
Script results (cleaned by now):
 
Tag: 
screen shot 2018-05-14 at 14 48 27
Pypi release:
screen shot 2018-05-14 at 14 43 01

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      - tests:
+          requires:
+            - dependencies
+
+      - deploy:
+          requires:
+            - tests
+          filters:
+            branches:
+              only: deploy-9
```
Comment (Anna-Livia) [inline .circleci/config.yml]: To be changed before merge

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+> This file defines the rules to follow to contribute to your repository.
+> The rules suggested here are what we generally use for OpenFisca packages.
+> You can of course edit them, and you should probably remove this block :)
```
Comment (Anna-Livia) [inline CONTRIBUTING.md]: I think this block should be removed :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+> This file defines the rules to follow to contribute to your repository.
```
Comment (Anna-Livia) [inline CONTRIBUTING.md]: I would use `guidelines` rather than `rules`

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      - save_cache:
+          key: v1-checkout-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - .
+            - ~/.ssh/known_hosts
+
+  dependencies:
+    working_directory: ~/extension-template
+    docker:
+      - image: circleci/python:2.7-jessie
```
Comment (Anna-Livia) [inline .circleci/config.yml:25]: why `-jessie`?

**Review by Anna-Livia — APPROVED** [review]

File: .circleci/config.yml (hunk 4 above)
Comment (bonjourmauko) [reply .circleci/config.yml:25]: Yup, no need to use jessie IMHO

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+### 1.1.5 - [#15](https://github.com/openfisca/extension-template/pull/15)
+
+* Add continuous deployment with CircleCI on merge on `master` branch
```
Comment (bonjourmauko) [inline CHANGELOG.md]: > Add continuous deployment with CircleCI on merge on `master` branch

Add continuous deployment with CircleCI,  **triggered by a** merge **to** `master` branch

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  > For instance :
+  > # 13.0.0 [#671](https://github.com/openfisca/openfisca-france/pull/671)
+  >
+  > ## 13.2.0 [#676](https://github.com/openfisca/openfisca-france/pull/676)
+  >
+  > ### 13.1.5 [#684](https://github.com/openfisca/openfisca-france/pull/684)
+
+- The second line indicates the type of the change. The possible types are:
+  - `Tax and benefit system evolution`: Calculation improvement, fix, or update. Impacts th
```
Comment (bonjourmauko) [inline CONTRIBUTING.md]: I think you have a formatting or copy/paste problem here.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  users interested in calculations.
+  - `Technical improvement`: Performances improvement, installing process change, formula synta
```
Comment (bonjourmauko) [inline CONTRIBUTING.md]: Same thing.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  change… Impacts the users who write legislation and/or deploy their own instance.
```
Comment (bonjourmauko) [inline CONTRIBUTING.md]: > … 

, etc.

**Review by bonjourmauko — APPROVED** [review]
Please do rebase before merging 👍

File: .circleci/config.yml (hunk 4 above)
Comment (sandcha) [reply .circleci/config.yml:25]: This is a pre-built circle-ci docker image according to [circle doc](https://circleci.com/docs/2.0/circleci-images/#python) > [docker hub](https://hub.docker.com/r/circleci/python/tags/).

File: CONTRIBUTING.md (hunk 6 above)
Comment (sandcha) [reply CONTRIBUTING.md]: 😶

**Pull request merged**